### DISCUSSION TITLE
Can I Stracture api folder in next js app to dpends on route -  like Nuxt js 3  style

### DISCUSSION THREAD

**Discussion by codehubdev123** [vercel/next.js#66656] [Help]
### Summary

Need use redtful api url format no need create nested folder 
I'm using Restful api format for my application i have like looks like 
exmple.com/api/product/{productId}/orders/{orderId}
question: Can i use better way to do it without created nexted folders
/products/[productid]/orders/[orderId]/route.ts	

**Example what i do in nuxt js 3**
 **One file per folder Catch all** 
~/server/api/products/[...].ts

**Than using all links like that** 
```
import { createRouter, defineEventHandler, useBase } from 'h3';
const router = createRouter();

router.get(
  '/',
  defineEventHandler((event) => {
    return 'URL  /product';
  })
);
router.post(
  '/:productsId:/orders/{orderId}',
  defineEventHandler((event) => {
    return 'URL /product/1/orders/2';
  })
);

router.post(
  '/:id/orders/',
  defineEventHandler((event) => {
    return 'URL /product/1/orders/';
  })
);



export default useBase('/api/products/', router.handler);
```

**Comment by JeelRajodiya** [comment]
Hello @codehubdev123, 

You can use the [catch-all-segnemts](https://nextjs.org/docs/app/building-your-application/routing/dynamic-routes#catch-all-segments) pattern in next.js 

I am highlighting that part of the docs below 





## Catch-all Segments

Dynamic Segments can be extended to **catch-all** subsequent segments by adding an ellipsis inside the brackets `[...folderName]`.

For example, `app/shop/[...slug]/page.js` will match `/shop/clothes`, but also `/shop/clothes/tops`, `/shop/clothes/tops/t-shirts`, and so on.

| Route                        | Example URL   | `params`                    |
| ---------------------------- | ------------- | --------------------------- |
| `app/shop/[...slug]/page.js` | `/shop/a`     | `{ slug: ['a'] }`           |
| `app/shop/[...slug]/page.js` | `/shop/a/b`   | `{ slug: ['a', 'b'] }`      |
| `app/shop/[...slug]/page.js` | `/shop/a/b/c` | `{ slug: ['a', 'b', 'c'] }` |
